nixos/modules/installer/kexec/kexec-boot.nix

```diff
@@ -0,0 +1,51 @@
+# This module exposes a config.system.build.kexecBoot attribute,
+# which returns a directory with kernel, initrd and a shell script
+# running the necessary kexec commands.
+
+# It's meant to be scp'ed to a machine with working ssh and kexec binary
+# installed.
+
+# This is useful for (cloud) providers where you can't boot a custom image, but
+# get some Debian or Ubuntu installation.
+
+{ pkgs
+, modulesPath
+, config
+, ...
+}:
+{
+  imports = [
+    (modulesPath + "/installer/netboot/netboot-minimal.nix")
+  ];
+
+  config = {
+    system.build.kexecBoot =
+      let
+        kexecScript = pkgs.writeScript "kexec-boot" ''
```

Comment:
Maybe add:

```nix
# this script intentionally uses system dependencies for efficiency,
# instead of e.g. ${bash}.
```

Would be cool to have a script that rolls everything up into a tarball or into a self-unpacking script, so this can deployed by copying one file or an `ssh foo 'curl | sh'` one-liner.
Portability would also be a reason not to hardcode those deps.

Reply:
* portability to systems without a Nix installation

Reply:
maybe we can take some inspiration here: https://github.com/nix-community/nixos-generators/blob/master/formats/kexec.nix#L23